src/js/models/LottoState.js

```diff
@@ -0,0 +1,85 @@
+import { PRIZE_FOR_PLACE } from '../constants.js';
+
+export default class LottoStateModel {
+  constructor() {
+    this.purchasedAmount = 0;
+    this.quantity = 0;
+    this.lottos = [];
+    this.isOpen = false;
+    this.places = {
+      place1: 0,
+      place2: 0,
+      place3: 0,
+      place4: 0,
+      place5: 0,
+    };
+    this.totalProfit = 0;
+  }
+
+  setIsOpen(openState) {
+    this.isOpen = openState;
+  }
+
+  setPurchasedAmount(amount) {
+    this.purchasedAmount = amount;
+  }
+
+  setQuantity(quantity) {
+    this.quantity = quantity;
+  }
+
+  getLottoNumbers() {
+    return this.lottos.map((lotto) => lotto.winningNumbers);
+  }
+
+  drawLotto(winningNumbers, bonusNumber) {
+    this.resetPlaces();
+    const winNumbers = winningNumbers.sort((a, b) => a - b);
+    const bonusNum = bonusNumber[0];
+    const lottos = this.getLottoNumbers();
+    lottos.forEach((lotto) => {
+      const res = lotto.filter((val) => winNumbers.includes(val));
+      const hasBonus = lotto.includes(bonusNum);
+      if (lotto.join('') === winNumbers.join('')) {
+        this.places['place1'] += 1;
+      } else if (res.length === 5 && hasBonus) {
+        this.places['place2'] += 1;
+      } else if (res.length === 5) {
+        this.places['place3'] += 1;
+      } else if (res.length === 4) {
+        this.places['place4'] += 1;
+      } else if (res.length === 3) {
+        this.places['place5'] += 1;
+      }
+    });
+
+    return this.places;
+  }
+
+  calculateProfitRatio() {
+    Object.keys(this.places).forEach((key) => {
+      this.totalProfit += PRIZE_FOR_PLACE[key] * this.places[key];
+    });
+
+    if (!this.totalProfit) return 0;
+    return (this.purchasedAmount / this.totalProfit) * 100;
+  }
+
+  initLottoState() {
+    this.purchasedAmount = 0;
+    this.quantity = 0;
+    this.lottos = [];
+    this.resetPlaces();
+    this.totalProfit = 0;
+  }
+
+  resetPlaces() {
+    this.places = {
+      place1: 0,
+      place2: 0,
+      place3: 0,
+      place4: 0,
+      place5: 0,
+    };
+  }
```

Comment:
```suggestion
  initLottoState() {
    this.purchasedAmount = 0;
    this.quantity = 0;
    this.lottos = [];
    this.#resetPlaces();
    this.totalProfit = 0;
  }

  #resetPlaces() {
    this.places = {
      place1: 0,
      place2: 0,
      place3: 0,
      place4: 0,
      place5: 0,
    };
  }
```
이렇게 외부에서 쓰이지 않는 것들은 `#`을 붙여서 private으로 접근지정을 제한할 수 있습니다!

Reply:
몰랐던 부분인데 알려주셔서 감사합니다!
[a92e7db](https://github.com/next-step/js-lotto/pull/184/commits/a92e7db080ea140910c9e81367ec13cd1ee28256)로 반영했습니다.